### PR TITLE
xdstp: flatten resource locators/names to strings.

### PR DESCRIPTION
**Pull request by htuch**
As per discussion summarized in
https://github.com/envoyproxy/envoy/pull/13555#issuecomment-736074440, we will not use structured
xdstp:// names/locators in the API initially. Instead, we will re-use existing string fields for
names and special case any name with a xdstp: prefix. We leave open the option of introducing
structured representation, in particular for efficiency wins, at a later point.

Risk level: Low (not in use yet)
Testing: CI

Signed-off-by: Harvey Tuch <htuch@google.com>

### CODE REVIEW

File: test/config/integration/server_xds.bootstrap.udpa.yaml [hunk 1]
```diff
@@ -1,8 +1,5 @@
 dynamic_resources:
-  lds_resources_locator:
-    scheme: FILE
-    id: {{ lds_json_path }}
-    resource_type: envoy.config.listener.v3.ListenerCollection
+  lds_resources_locator: file:///{{ lds_json_path }}
```
Comment (htuch) [inline test/config/integration/server_xds.bootstrap.udpa.yaml:2]: Note that we need to distinguish flat string that start with `file:`, `http:` as well as `xdstp:` going forward.

File: api/envoy/service/discovery/v3/discovery.proto [hunk 2]
```diff
@@ -243,11 +227,6 @@ message DeltaDiscoveryResponse {
   // Removed resources for missing resources can be ignored.
   repeated string removed_resources = 6;
 
-  // As with *removed_resources* but used when a removed resource was named in
-  // its *Resource*s with a *xds.core.v3.ResourceName*.
-  // [#not-implemented-hide:]
-  repeated xds.core.v3.ResourceName udpa_removed_resources = 7;
```
Comment (markdroth) [inline api/envoy/service/discovery/v3/discovery.proto:249]: Please also remove the comment on line 222 that refers to this field.

**Review by markdroth — DISMISSED** [review]
Looks great!

**Review by mattklein123 — APPROVED** [review]
Thanks, I think this is the right call for the ecosystem.

**Pull request merged**